CCVA_Processing.Rmd

```diff
@@ -587,20 +620,91 @@ annual_stats <- centroid_annual %>%
   dplyr::summarize_all(.funs = 
                         list(mean = ~ mean(., na.rm = TRUE),
                           median = ~ median(., na.rm = TRUE),
-                              sd = ~ sd(., na.rm = TRUE)))
+                              sd = ~ sd(., na.rm = TRUE)),
+                       .groups = "keep")
+
+annual_percent <- (100*(annual_stats[2:5,3:ncol(annual_stats)] -           
+                       annual_stats[rep(1,4),3:ncol(annual_stats)]) / 
+                       annual_stats[rep(1,4),3:ncol(annual_stats)]) %>%
+                      cbind(annual_stats[2:5,1:2],.)
   
-seasonal_stats <- centroid_seasonal %>%
-  dplyr::mutate(stat_group = ifelse(y > hist_years[1] & y < hist_years[2], "Historical",
-                             ifelse(y > future_years[1] & y < future_years[2], 
-                                    "Future",NA))) %>%
-  dplyr::ungroup() %>%
-  tidyr::drop_na(any_of("stat_group")) %>%
-  dplyr::group_by(cf,stat_group,season) %>%
-  dplyr::summarize_all(.funs = 
-                        list(mean = ~ mean(., na.rm = TRUE),
-                          median = ~ median(., na.rm = TRUE),
-                              sd = ~ sd(., na.rm = TRUE)))
 
+# KEC - there is a bimodal distribution for the day-of-year for the maximum
+# precipitation AND correspondingly runoff. 
+
+# Group by year and select daily maximum discharge with corresponding date
+max_annual_runoff <- centroid_all %>%
+  mutate(Date = date,
+         year = year(date)) %>%
+  dplyr::select(c(Date,cf,year,runoff_in)) %>%
+  addWaterYear() %>%
+  dplyr::group_by(cf,waterYear) %>% 
+  dplyr::slice_max(order_by = runoff_in, n=1, with_ties = FALSE) %>%
+  mutate(doy = yday(Date),
+         dowy = as.numeric(Date - as.Date(paste0(waterYear-1,"-10-01"))),
+         season = case_when(month(Date) <= 5 ~ "spring",
```

Comment:
Just clarifying the season identification here, is this common practice in the field to not include winter? And also summer is just June and July?